### PR TITLE
Added new extension primitives

### PR DESCRIPTION
**Pull request by jeluard**
Added move components for extensions

status: ready

### CODE REVIEW

File: src/status_im/utils/http.cljs [hunk 1]
```diff
@@ -2,13 +2,38 @@
...
+  ([url body on-success on-error {:keys [timeout-ms]}]
+   (-> (rn-dependencies/fetch url
+                              (clj->js {:method  "POST"
+                                        :headers {"Cache-Control" "no-cache"}
+                                        :body    body
+                                        :timeout (or timeout-ms http-request-default-timeout-ms)}))
+       (.then (fn [response]
+                (->
+                 (.text response)
+                 (.then (fn [body]
```
Comment (bhauman) [inline src/status_im/utils/http.cljs:29]: does `.text` return a promise?

Comment (jeluard) [reply src/status_im/utils/http.cljs:29]: Yes it does

**Review by flexsurfer — APPROVED** [review]

**Comment by alwx** [comment]
Can we not merge it before this one? https://github.com/status-im/status-react/pull/6341
Just because we need to have #6341 merged asap and otherwise we will also need to rebase it and maybe even check it again first.
Thanks.

File: src/status_im/utils/http.cljs [hunk 2]
```diff
@@ -2,13 +2,38 @@
...
-            [goog.Uri :as uri]
             [clojure.string :as string])
   (:refer-clojure :exclude [get]))
 
 ;; Default HTTP request timeout ms
 (def http-request-default-timeout-ms 3000)
 
+(defn- headers [response]
+  (let [entries (es6-iterator-seq (.entries (.-headers response)))]
+    (reduce #(assoc %1 (string/trim (string/lower-case (first %2))) (string/trim (second %2))) {} entries)))
```
Comment (alwx) [inline src/status_im/utils/http.cljs:13]: it looks a bit messy and can be simplified, but seems like it's not really related to this pull request

**Review by alwx — APPROVED** [review]

**Pull request merged**